### PR TITLE
Sara/release notes/aug2023

### PR DESCRIPTION
**Pull request by s-santillan**
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [x] A subject matter expert (SME) reviews the content
- [x] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs
- [x] If you have changed any header tag links (doc/#this-kind-of-anchor), update all instances of that link

This PR includes the new endpoint /latest

latest-endpoint

### CODE REVIEW

File: docs/release-notes/august-2023.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+### Removed
+
+- Removed the ability to turn off scanning with lockfile-only rules. Moving foward, lockfile-only rules are included in all full scans.
+
+## Semgrep Assistant
+
+### Fixed
+
+- GitHub: Fixed a bug in which you can receive duplicate PR comments if they had installed more than one instance of `semgrep-app`.
```
Comment (armchairlinguist) [inline docs/release-notes/august-2023.md:153]:
```suggestion
- GitHub: Fixed a bug in which you ciykd receive duplicate PR comments if you had installed more than one instance of `semgrep-app`.
```

File: docs/release-notes/august-2023.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+description: >-
+  Release notes include the changes, fixes, and additions in specific versions of Semgrep.
+---
+
+# August 2023 release notes
+
+## Semgrep OSS Engine
+
+:::caution
+Semgrep version 1.38.0 introduces several removals that may break your Semgrep workflows. See [Semgrep OSS > Removed](#removed) for more information.
```
Comment (armchairlinguist) [inline docs/release-notes/august-2023.md:15]: "introduces several removals" is strange. "Removes some functionality", maybe?

File: docs/release-notes/august-2023.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+- `metavariable-comparison`: You can now use `in` and `not in` for strings in the same sense as in Python, for substring checking. ([#2979](https://github.com/returntocorp/semgrep/pull/8406))
+- **Julia:** Added the deep expression operator. Now you can write patterns such as `foo(<... 42 ...>)` to find instances of calls to `foo` that contain `42` somewhere inside of it. ([#8540](https://github.com/returntocorp/semgrep/pull/8540))
+- Added support for languages with case insensitive identifiers and generalized PHP to use these case insensitive identifiers.
+  - For example, in PHP the pattern `MyClass()` now matches calls with different capitalizations such as `myclass()` and `Myclass()`. ([#8356](https://github.com/returntocorp/semgrep/pull/8356))
+
+### Fixed
+
+- Fixed `--text` and `--output` flags which broke in 1.38.0. If you are using version 1.38.0, update Semgrep to receive these fixes.
+- Version 1.38.0 introduced some regressions with the Semgrep docker image. This is fixed in version 1.38.3.
```
Comment (armchairlinguist) [inline docs/release-notes/august-2023.md:34]: This phrasing seems odd to me, because technically there were never regressions with the version tagged 1.38.0 - the problem was with the `latest` tag behavior. But I'm having a hard time coming up with anything better. `Docker` does need to be capitalized though I think.

File: docs/release-notes/august-2023.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+- Converted all '@r2c.dev' email addresses to '@semgrep.com'. Several error messages displayed outdated email addresses. With this fix, you can now see the correct email to reach out to the Semgrep Support team at [support@semgrep.com](mailto:support@semgrep.com). ([#8446](https://github.com/returntocorp/semgrep/pull/8446))
+- Fixed CLI output to display matches from different rules with the same message. Now you are able to see the rule ID granularly even if two rules have the same rule message. ([#8557](https://github.com/returntocorp/semgrep/pull/8557))
+- Semgrep PyPI package can now be installed on **aarch64 libmusl** platforms such as Alpine. (gh-8565)
+- Improved the `--max-memory` help description to make it clearer. Its previous message, "Defaults to 0 for all CLI scans" implies a different default for non-CLI scans, where in practicality the default is 0 for all scans except when using Pro Engine, where the default is 5000.
```
Comment (armchairlinguist) [inline docs/release-notes/august-2023.md:38]: This description is itself not quite right, I believe. @sinat101 Can you provide more context if I'm missing something? The default remains 0 for Pro Engine scans on CLI (just ran one myself on 1.38.3 to check). So presumably there are only certain environments where the 5000 default kicks in.

File: docs/release-notes/august-2023.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+      }
+  
+  }
+  ```
+  Semgrep was considering that `this.x` referred to the parameter `x` of `test2` rather than to the class attribute `x`. ([#8508](https://github.com/returntocorp/semgrep/pull/8508))
+
+
+### Changed
+
+- Running the `semgrep`  command now displays the help message. Previously, the `semgrep` command ran a SAST scan.
```
Comment (armchairlinguist) [inline docs/release-notes/august-2023.md:95]:
```suggestion
- Running the `semgrep`  command with no subcommand now displays the help message. Previously, the `semgrep` command ran a SAST scan by default.
```

File: docs/release-notes/august-2023.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+### Fixed
+
+- **JavaScipt (JS) or TypeScript (TS) taint mode:** fixed a bug introduced in 1.33.1 that had the side-effect of hurting performance of taint rules on JS or TS repositories that used destructuring in a function's formal parameters.
+
+## Semgrep Cloud Platform
+
+### Added
+
+- The `semgrep ci` command now displays enabled products when scans are command now created or when the scan config is generated from Semgrep Cloud Platform. Additionally, if no products are enabled then a friendly error is raised and the scan is stopped. You must enable a product to start a scan.
```
Comment (armchairlinguist) [inline docs/release-notes/august-2023.md:120]:
```suggestion
- The `semgrep ci` command now displays enabled products when the scan config is generated from Semgrep Cloud Platform. Additionally, if no products are enabled then a friendly error is raised and the scan is stopped. You must enable a product to start a scan.
```

It seems weird to me to have this in the SCP section since it's a change in Semgrep behavior, even if it only affects `semgrep ci`. Thoughts?

Comment (s-santillan) [reply docs/release-notes/august-2023.md:120]: Just curious - what seems weird about it to you? My thought was: `semgrep ci` can run any, all, or none of the products, but does require an SCP account and isn't under Semgrep OSS.

Comment (armchairlinguist) [reply docs/release-notes/august-2023.md:120]: That seems reasonable. It stuck me as odd, but logically I think your reasoning is better than my instinct 😄

File: docs/release-notes/august-2023.md (hunk 3 above)
Comment (s-santillan) [reply docs/release-notes/august-2023.md:34]: I'll remove this line - the message to UPDATE NAO is also served by the breakage in the output and text flags.

File: docs/release-notes/august-2023.md (hunk 4 above)
Comment (s-santillan) [reply docs/release-notes/august-2023.md:38]: @armchairlinguist I tried to check this for myself today - what command were you running to display the --max-memory? because I ran on verbose and nothing appeared 🤯

Comment (armchairlinguist) [reply docs/release-notes/august-2023.md:38]: Ah yes sorry, I should have said - it requires using --debug. I only ran it long enough for the semgrep-core invocation, since that's where you see which arguments get passed in.

Comment (s-santillan) [reply docs/release-notes/august-2023.md]: We got to the bottom of this case! Resolving.

File: docs/release-notes/august-2023.md [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+
+- **JavaScipt (JS) or TypeScript (TS) taint mode:** fixed a bug introduced in 1.33.1 that had the side-effect of hurting performance of taint rules on JS or TS repositories that used destructuring in a function's formal parameters.
+
+## Semgrep Cloud Platform
+
+### Added
+
+- The `semgrep ci` command now displays enabled products when scans are command now created or when the scan config is generated from Semgrep Cloud Platform. Additionally, if no products are enabled then a friendly error is raised and the scan is stopped. You must enable a product in Semgrep **Cloud Platform > Settings** to start a scan.
+- You can now remove your SSO configuration. Previously, you had to reach out to [support@semgrep.com](mailto:support@semgrep.com) to remove SSO configurations.
+- **Projects page:** Added a **Sync projects** button which enables you to synchronize your Semgrep projects with your SCM. This means you are able to see the current status of your scans across all your projects that you have onboarded to Semgrep Cloud Platform.
```
Comment (annikalee) [inline docs/release-notes/august-2023.md:121]: "This means you are able to see the current status of your scans across all your projects that you have onboarded to Semgrep Cloud Platform." -> @ThatDevShparki doesn't this technically just mean that the projects will be there and it's more for enabling easier onboarding of additional projects? Being able to see the status was already there?

Comment (ThatDevShparki) [reply docs/release-notes/august-2023.md:121]: That's correct- the sync button just allows you to manually synchronize your current repository list with those available in the SCMs where those repositories originated from

Comment (annikalee) [reply docs/release-notes/august-2023.md:121]: @s-santillan I think we should rephrase this to say "Added a **Sync projects** button which enables you to synchronize your Semgrep projects with your SCM. This will enable you to faster onboard projects to the Semgrep Cloud Platform and ensure all your repositories are represented and available for scanning." (or something similar)

Comment (s-santillan) [reply docs/release-notes/august-2023.md:121]: Sounds perfect. Thank you @annikalee and @ThatDevShparki

Comment (s-santillan) [reply docs/release-notes/august-2023.md]: Resolving as fixed in latest commit.

**Review by armchairlinguist — APPROVED** [review]
It'd be great to incorporate these last comments but I'll approve so it's not blocked as it's not incredibly critical stuff 👍🏻

File: docs/release-notes/august-2023.md [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+- Converted all '@r2c.dev' email addresses to '@semgrep.com'. Several error messages displayed outdated email addresses. With this fix, you can now see in the CLI the correct email to reach out to the Semgrep Support team, which is [support@semgrep.com](mailto:support@semgrep.com). ([#8446](https://github.com/returntocorp/semgrep/pull/8446))
+- Fixed CLI output to display matches from different rules with the same message. Now you are able to see the rule ID granularly even if two rules have the same rule message. ([#8557](https://github.com/returntocorp/semgrep/pull/8557))
+- Semgrep PyPI package can now be installed on **aarch64 libmusl** platforms such as Alpine. (gh-8565)
+- Improved the `--max-memory` help description to make it clearer. Its previous message, "Defaults to 0 for all CLI scans," did not convey that the default is 0 for all scans except when using Semgrep Pro Engine in CI scans. The default is 5000MiB for Semgrep Pro Engine CI scans. In this context, a CI scan is any scan using the `semgrep ci` command, whether in a local environment or a CI/CD pipeline, and Pro Engine must be enabled for the org whose repositories you are scanning.
```
[the remark this answers is not in the document]
Comment (s-santillan) [reply docs/release-notes/august-2023.md:38]: In that case, I think I can add the --flag as well :) It's good to know.

**Comment by s-santillan** [comment]
I'll merge on Monday at 10AM PST, if any notes come after that, I will make a follow-up PR.

File: docs/release-notes/august-2023.md [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+
+:::caution
+Semgrep version 1.38.0 removed some features that may break your Semgrep workflows. See [Semgrep OSS > Removed](#removed) for more information.
+:::
+
+This section of release notes includes upgrades of Semgrep OSS Engine for versions between **1.35.0** and **1.38.3**.
+
+### Added
+
+- The CLI now returns the commit timestamp when running `semgrep ci`.
```
[the remark this answers is not in the document]
Comment (s-santillan) [reply docs/release-notes/august-2023.md]: I changed it to collects, just in case.
